spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/functions/IcebergVersionFunction.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.functions;
+
+import org.apache.iceberg.IcebergBuild;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.catalog.functions.BoundFunction;
+import org.apache.spark.sql.connector.catalog.functions.ScalarFunction;
+import org.apache.spark.sql.connector.catalog.functions.UnboundFunction;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.unsafe.types.UTF8String;
+
+/** Implementation of {@link UnboundFunction} that prints the current Iceberg version in use. */
+public class IcebergVersionFunction implements UnboundFunction {
+  @Override
+  public BoundFunction bind(StructType inputType) {
+    if (inputType.fields().length > 0) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "Invalid input type. Expected 0 fields but found %s", inputType.fields().length));
+    }
+
+    return new IcebergVersion();
+  }
+
+  @Override
+  public String description() {
+    return "iceberg_version - Returns the runtime Iceberg version for debugging purposes";
+  }
+
+  @Override
+  public String name() {
+    return "iceberg_version";
+  }
+
+  public static class IcebergVersion implements ScalarFunction<String> {
+
+    private static final UTF8String icebergBuildVersion =
+        UTF8String.fromString(IcebergBuild.version());
+
+    public static UTF8String invoke() {
+      return icebergBuildVersion;
+    }
+
+    @Override
+    public DataType[] inputTypes() {
+      return new DataType[0];
+    }
+
+    @Override
+    public DataType resultType() {
+      return DataTypes.StringType;
+    }
+
+    @Override
+    public String canonicalName() {
+      return "org.apache.iceberg.spark.functions.iceberg_version()";
```

Comment:
How about `iceberg.iceberg_version`?

Reply:
Sure.

Spark uses these to differentiate the identity of two functions across catalogs and namespaces, so I erred on the longer side.

But I think `iceberg.` is fine as this isn’t really meant for end users anyway and if some fork maintainer is using `iceberg.iceberg_version` for different logic I’d be very surprised.

Reply:
Yeah, we do want to make sure that the string is unique. But there is also no need to make it overly complicated.